src/job.ts

```diff
@@ -296,8 +309,16 @@ export class CronJob<OC extends CronOnCompleteCommand | null = null, C = null> {
 	stop() {
 		if (this._timeout) clearTimeout(this._timeout);
 		this.running = false;
-		if (typeof this.onComplete === 'function') {
-			void this.onComplete.call(this.context);
-		}
+
+		const waitForCompletion = async () => {
```

Comment:
Similar to the `fireOnTick` comment, this breaks existing behavior and should be controlled by a parameter in the function signature, which would be set to `false` by default.